libr/include/r_types.h

```diff
@@ -13,6 +13,20 @@
 #undef __UNIX__
 #undef __WINDOWS__
 
+#define R_IN /* do not use, implicit */
+#define R_OWN /*own*/
```

Comment:
I guess that when it's used on a return value of a function, it means that the caller is responsible for freeing the element.

What about when it's used as argument to a function? What does that mean? Shall we be more descriptive in the comment to document the behaviour? Currently I'm not sure I understand 100% this one

Reply:
The comments should be more descriptive. I'll assume `R_OWN` as argument annotation means the callee will take ownership of whatever the argument owns, resembling the following:
```cpp
void sink(std::unique_ptr<int> p)
void sink(std::vector<int>&& p)
```

Reply:
btw, what about R_OWNED instead of R_OWN